syntax = "proto3";
package upload;

message DocumentUploadRequest {
  oneof data {
    DocumentMetaData meta_data = 1;
    bytes chunk_data = 2;
    bool end_signal = 3;  // Signals upload completion
  }
}

message DocumentMetaData {
  string file_name = 1;
  string description = 4;
  string date_captured = 5;
}

message DocumentUploadResponse {
  bool success = 1;
  string message = 2;
  int32 chunk_number = 3;  // For progress tracking
}